### ISSUE TITLE
Update PLT when list of child dependencies changes

### ISSUE THREAD

**Issue by mguimas** [jeremyjh/dialyxir#232]
Happens with Elixir 1.7.2 and Dialyxir 0.5.1

- I have an umbrella project with several child projects;

- The umbrella and the child projects have the following in their `mix.exs` files (and nothing else regarding `dialyxir`):

```
defp deps do
    [
      ...
      {:dialyxir, "~> 0.5", only: [:dev, :test]}
      ...
    ]
end
```

- Added a new dependency in one of the child projects (this dependency did not exist before in any child project).

Then,

**Observed behavior**: running `mix dialyzer` inside the child project does not update the PLT, but running the same command at the umbrella project updates the PLT.

**Expected behavior**: when child dependencies change, running `mix dialyzer` inside the corresponding child project, or running at the umbrella project, should update the PLT. The PLT check should always run, unless `mix dialyzer --no-check` is run.

**Comment by jeremyjh** [comment]
Umbrella projects share a build directory and a lock file, so they also share a PLT. Child projects reference the parents `mix.lock` file, so when that is changed a PLT check will run. I wonder, did you use mix to generate your child apps from inside the umbrella project? 

Just running 
```shell
mix new project_name --umbrella
cd project_name/apps
mix new child1
```

Gives me a project config with these keys in the child: 

```elixir
    build_path: "../../_build",
      config_path: "../../config/config.exs",
      deps_path: "../../deps",
      lockfile: "../../mix.lock",
```
After that I ran dialyzer, then added a dep in child1 and ran it again - it updated the PLT. Maybe the issue you see is that it will not check the PLT unless you run `mix dialyzer` from the command line? It does tell you this in the output:  `In an Umbrella child, not checking PLT...`

I'm not terribly happy with the way Umbrella dependency resolution is implemented, but since I do not use umbrella projects I'm unlikely to invest a lot more effort into solving it.

**Comment by mguimas** [comment]
Yes, I saw the message `In an Umbrella child, not checking PLT...` when running from inside the child project, and when running from the umbrella project dir I saw `Checking PLT...` (and only this time the PLT was updated).

Updating the PLT when running from inside a child project has the benefit of running Dialyzer only for that child, which is faster than running for all children when calling `mix dialyzer` from the umbrella project dir; also, it only shows the type errors for that child, not all children.

If this is your policy "since I do not use umbrella projects I'm unlikely to invest a lot more effort into solving it", then you should **document clear** that `dialyxir` have some limitations when used in umbrella projects and state them :-(

**Comment by mguimas** [comment]
ok, then let the issue stay open for a PR. Let me see if I can live with it, otherwise I may submit a PR.
Thanks.

**Comment by harlantwood** [comment]
@jeremyjh first thanks for this awesome tool!  

As a workaround for this issue, I am doing this from a script that runs in each subproject of my umbrella project:

```
pushd ../.. && mix dialyzer --plt && popd
mix dialyzer --halt-exit-status
```

This should mean I'm always updating the PLT for each subproject, as I understand it.

**Comment by jeremyjh** [comment]
@harlantwood on CI you could just run dialyzer from the umbrella root, right? Its only an issue when you want to run dialyzer from the child.

**Comment by harlantwood** [comment]
We run a separate build on CI for each child project, and we'd like to see dialyzer errors there that are just for that child project.
